ee/src/main/java/org/jboss/as/ee/concurrent/ControlPointUtils.java

```diff
@@ -212,12 +215,18 @@ public T call() throws Exception {
                     if (controlPoint.beginRequest() == RunResult.RUN) {
                         try {
                             return callable.call();
+                        } catch (Exception e) {
+                            throw new ExecutionException(e);
                         } finally {
                             controlPoint.requestComplete();
                         }
                     }
+                } catch (ExecutionException e) {
+                    // from callable
+                    throw (Exception) e.getCause();
                 } catch (Exception e) {
-                    EeLogger.ROOT_LOGGER.failedToRunTask(e);
+                    // from controlPoint.beginRequest(), Callable mandates exception
+                    throw EeLogger.ROOT_LOGGER.failedToProcureCallableResult(callable, e);
```

Comment:
This should use failureWhileRunningTask(...).
We don't want to throw a checked exception if ControlPoint.beginRequest() threw an exception.

Reply:
Thats Callable contract.

Reply:
We only want to propagate a checked exception if the underlying Callable.call() throws a checked exception.  If this method were to throw a checked exception due to ControlPoint.beginRequest() failure - that would not comply with the Callable contract.